python_modules/dagster/dagster/_core/execution/submit_asset_runs.py

```diff
@@ -142,47 +148,58 @@ def _create_asset_run(
                 workspace=workspace,
                 run_request_execution_data_cache=run_request_execution_data_cache,
             )
-            check_for_debug_crash(debug_crash_flags, "EXECUTION_PLAN_CREATED")
-            check_for_debug_crash(debug_crash_flags, f"EXECUTION_PLAN_CREATED_{run_request_index}")
 
-            # retry until the execution plan targets the asset selection
-            if _execution_plan_targets_asset_selection(
-                execution_data.external_execution_plan.execution_plan_snapshot,
+        except (DagsterInvalidSubsetError, DagsterUserCodeProcessError):
+            logger.warning(
+                "Error while generating the execution plan, possibly because the code server is out of sync with the daemon. Pausing run submission for "
+                f"{RELOAD_WORKSPACE_INTERVAL} to allow the execution plan to rebuild with the updated workspace.",
+                exc_info=sys.exc_info(),
+            )
+            should_retry = True
```

Comment:
Should we not provide more context about the exception itself?

Reply:
The `exc_info=sys.exc_info()` does this. Sample log line:
```
WARNING:dagster.daemon.BackfillDaemon:Error while generating the execution plan, possibly because the code server is out of sync with the daemon. Pausing run submission for 60 to allow the execution plan to rebuild with the updated workspace.
Traceback (most recent call last):
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/execution/submit_asset_runs.py", line 144, in _create_asset_run
    execution_data = _get_job_execution_data_from_run_request(
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/execution/submit_asset_runs.py", line 94, in _get_job_execution_data_from_run_request
    external_job = code_location.get_external_job(pipeline_selector)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/remote_representation/code_location.py", line 150, in get_external_job
    subset_result = self.get_subset_external_job_result(selector)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/remote_representation/code_location.py", line 821, in get_subset_external_job_result
    subset = sync_get_external_job_subset_grpc(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_api/snapshot_job.py", line 47, in sync_get_external_job_subset_grpc
    raise DagsterUserCodeProcessError.from_error_info(result.error)
dagster._core.errors.DagsterUserCodeProcessError: dagster._core.errors.DagsterInvalidSubsetError: AssetKey(s) ['hourly_asset'] were selected, but no AssetsDefinition objects supply these keys. Make sure all keys are spelled correctly, and all AssetsDefinitions are correctly added to the `Definitions`.

Stack Trace:
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_grpc/impl.py", line 283, in get_external_pipeline_subset_result
    definition = repo_def.get_maybe_subset_job_def(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/definitions/repository_definition/repository_definition.py", line 330, in get_maybe_subset_job_def
    return defn.get_subset(
           ^^^^^^^^^^^^^^^^
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/definitions/job_definition.py", line 760, in get_subset
    return self._get_job_def_for_asset_selection(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/definitions/job_definition.py", line 786, in _get_job_def_for_asset_selection
    job_asset_graph = get_asset_graph_for_job(self.asset_layer.asset_graph, selection)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/definitions/asset_job.py", line 290, in get_asset_graph_for_job
    selected_keys = selection.resolve(parent_asset_graph)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/definitions/asset_selection.py", line 420, in resolve
    return self.resolve_inner(asset_graph, allow_missing=allow_missing)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dgibson/dagster/python_modules/dagster/dagster/_core/definitions/asset_selection.py", line 897, in resolve_inner
    raise DagsterInvalidSubsetError(
```